packages/core/core-flows/src/order/utils/maybe-refresh-shipping-methods.ts

```diff
@@ -0,0 +1,170 @@
+  "shipping_profile_id",
+  "provider_id",
+  "data",
+
+  "type.id",
+  "type.label",
+  "type.description",
+  "type.code",
+
+  "provider.id",
+  "provider.is_enabled",
+
+  "rules.attribute",
+  "rules.value",
+  "rules.operator",
+]
+
+/**
+ * The data to create a shipping method for an order edit.
+ */
+export type MaybeRefreshShippingMethodsWorkflowInput = {
+  /**
+   * The ID of the shipping method to refresh.
+   */
+  shipping_method_id: string
+  /**
+   * The ID of the order.
+   */
+  order_id: string
+  /**
+   * The ID of the ADD SHIPPING action to update.
+   */
+  action_id: string
+  /**
+   * Data to pass for the shipping calculation.
+   */
+  context: CalculatedRMAShippingContext
+}
+
+export const maybeRefreshShippingMethodsWorkflowId =
+  "maybe-refresh-shipping-methods"
+/**
+ * This workflows refreshes shipping method an order (used in RMA flows).
+ * The shipping method and the action is updated if the shipping option is calculated.
+ */
+export const maybeRefreshShippingMethodsWorkflow = createWorkflow(
+  maybeRefreshShippingMethodsWorkflowId,
+  function (
+    input: MaybeRefreshShippingMethodsWorkflowInput
+  ): WorkflowResponse<void> {
+    const shippingMethod = useRemoteQueryStep({
+      entry_point: "order_shipping_method",
+      fields: ["id", "shipping_option_id"],
+      variables: {
+        id: input.shipping_method_id,
+      },
+      list: false,
+    }).config({ name: "fetch-shipping-method" })
+
+    const shippingOption = useRemoteQueryStep({
+      entry_point: "shipping_option",
+      fields: [...COMMON_OPTIONS_FIELDS],
+      variables: { id: shippingMethod.shipping_option_id },
+      list: false,
+    }).config({ name: "calculated-option" })
+
+    const isCalculatedPriceShippingOption = transform(
+      shippingOption,
+      (option) => option?.price_type === ShippingOptionPriceType.CALCULATED
+    )
+
+    when(
+      { isCalculatedPriceShippingOption, shippingOption },
+      ({ isCalculatedPriceShippingOption, shippingOption }) =>
+        isCalculatedPriceShippingOption
+    ).then(() => {
+      const order = useRemoteQueryStep({
+        entry_point: "orders",
+        fields: ["id", "shipping_address", "items.*", "items.variant.*"],
+        variables: { id: input.order_id },
+        list: false,
+        throw_if_key_not_found: true,
+      }).config({ name: "order-query" })
+
+      const calculateShippingOptionsPricesData = transform(
+        {
+          shippingOption,
+          order,
+          input,
+        },
+        ({ shippingOption, order, input }) => {
+          return [
+            {
+              id: shippingOption.id as string,
+              optionData: shippingOption.data,
+              context: {
+                ...order,
+                ...input.context,
+                from_location:
+                  shippingOption.service_zone.fulfillment_set.location,
+              },
+              // data: {}, // TODO: add data
```

Comment:
question: What should we do here?

Reply:
Atm we are not handling fulfillment data validation for shipping methods across the RMA flows, we just copy shipping option data. Should we revisit this separately?

Reply:
Let's revisit this separately. Can you create a ticket for this?

Reply:
yup, created CMRC-987